File: robertly-ui/src/app/models/series-per-muscle.ts
export interface SeriesPerMuscleRow {
  muscleGroup: string | null;
  year: number | null;
  month: number | null;
  week: number | null;
  totalSeries: number | null;
  firstDateInPeriod: string | null;
  targetValue: number | null;
}

export interface SeriesPerMuscle {
  seriesPerMuscleWeekly: SeriesPerMuscleRow[];
  seriesPerMuscleMonthly: SeriesPerMuscleRow[];
  seriesPerMuscleYearly: SeriesPerMuscleRow[];
}

File: robertly-net-api/api/Models/SeriesPerMuscle.cs
using System;
using System.Collections.Generic;

namespace robertly.Models;

public record SeriesPerMuscleRow
{
  public required string MuscleGroup { get; init; }
  public required int Year { get; init; }
  public required int Month { get; init; }
  public required int Week { get; init; }
  public required int TotalSeries { get; init; }
  public int TargetValue { get; set; }

}

public record SeriesPerMuscle
{
  public required IEnumerable<SeriesPerMuscleRow> SeriesPerMuscleWeekly { get; init; } = [];
  public required IEnumerable<SeriesPerMuscleRow> SeriesPerMuscleMonthly { get; init; } = [];
  public required IEnumerable<SeriesPerMuscleRow> SeriesPerMuscleYearly { get; init; } = [];
}